pkg/tracing/jaeger/config_yaml.go

```diff
@@ -164,6 +164,30 @@ func getRemoteOptions(config Config) []jaegerremote.Option {
 	return remoteOptions
 }
 
+func getRateLimitingOptions(config Config) []jaegerremote.Option {
```

Comment:
@matej-g this is the first draft of rate limiting sampler options.
This assumes the strategy will be parsed from the endpoint, which is polled regularly.

Regarding the maximum number of traces per second, I was considering adding another option to the config or reading a JSON file, and then posting that in JSON format to the URL. This way, the user can add that too to one config, instead of having a separate config on the URL.
Please let me know what you think, not entirely sure when it comes to remote and rate limited samplers 😅 

Final leg of the PR, we got this 💪🏻 

Thanks!

Reply:
Thanks Aditi! Looking good with remote sampler, but I have a feeling we had misunderstood each other regarding rate limiting sampler.

rate limiting sampler =/= remote sampler

Rate limiting sampler should not need to contact a remote endpoint, it should work based on a configuration provided to Thanos. Example from existing version Thanos:
```
  "sampler_param": 2
  "sampler_type": "ratelimiting"
  "service_name": "thanos-receive"
```

This will allow only 2 traces per second. So this sampler is "static" and is not controlled by remote endpoint. I probably did not explain this correctly during last F2F - that's why we need to find a way to introduce this sampler, because I believe this sampler type _by itself_ might not yet exist in OTEL. Basically we would need this sampler https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/samplers/jaegerremote/sampler.go#L115 as a stand-alone sampler.

Reply:
@matej-g thanks!
I have added an implementation of rate limiting sampler, based on related OTel-go code. I had to introduce quite a few of the functions required since rate-limiting samplers related utilities are `internal` to OTel-go.
I _think_ this is the only way that strikes me now to introduce rate limiting samplers. It might be a temporary, stop-gap arrangement until the required utilities are exported in future by OTel-go. 
Please let me know what you think.
thanks!